pkg/doc/signature/ed25519signature2018/suite.go

```diff
@@ -52,7 +52,7 @@ func (s *SignatureSuite) GetCanonicalDocument(doc map[string]interface{}) ([]byt
 
 // GetDigest returns document digest
 func (s *SignatureSuite) GetDigest(doc []byte) []byte {
-	digest := sha512.Sum512(doc)
+	digest := sha256.Sum256(doc)
```

Comment:
What is the reason for this change? Where is the test that validates this change?

Reply:
Please see https://w3c-dvcg.github.io/ld-signatures/#create-verify-hash-algorithm, ISSUE 8:
> The Ed25519Signature2018 algorithm also does not perform this last step -- and, in fact, uses SHA-512 internally. In short, this last step should better communicate that the 64 bytes produced from concatenating the SHA-256 of the canonicalized options with the SHA-256 of the canonicalized document are passed into the signing algorithm with a presumption that the signing algorithm will include hashing of its own.

We will call GetDigest() twice and concatenate the result, so it's worth to have it 32+32 bytes.

Reply:
Here is how it's implemented in https://github.com/digitalbazaar/jsonld-signatures:

https://github.com/digitalbazaar/jsonld-signatures/blob/f1e78c2674a786c3cdc8d310dbf228ed8a1e1b8e/lib/suites/LinkedDataSignature.js#L213